File: Frontend/trackr-administrador/src/app/shared/models/gestion-entidad/seccion-campo.ts
import { Dominio } from "@models/catalogo/dominio";
import { DominioDetalle } from "@models/catalogo/dominio-detalle";
import { Seccion } from './seccion';

export class SeccionCampo {

    public idSeccionCampo: number;
    public clave: string;
    public descripcion: string;
    public idDominio: number;
    public idSeccion: number;
    public requerido: boolean;
    public orden: number;
    public tamanoColumna: number;
    public habilitado: boolean;
    public valor?: number[] | boolean | string | Date | number;
    public grupo: string;
    public fila: number;
    public idIcono: number;
    public mostrarDashboard : boolean;

    // Extras - Utilerias
    public idDominioNavigation: Dominio;
    public listaOpciones: DominioDetalle[];
    public idEntidadEstructuraValor: number;
    public idSeccionNavigation : Seccion;

    constructor() {}
}

File: Backend/TrackrAPI/TrackrAPI/Models/SeccionCampo.cs
using System;
using System.Collections.Generic;

namespace TrackrAPI.Models
{
    public partial class SeccionCampo
    {
        public SeccionCampo()
        {
            EntidadEstructuraTablaValor = new HashSet<EntidadEstructuraTablaValor>();
        }

        public int IdSeccionCampo { get; set; }
        public string Clave { get; set; } = null!;
        public string Descripcion { get; set; } = null!;
        public int IdDominio { get; set; }
        public int IdSeccion { get; set; }
        public bool Requerido { get; set; }
        public int Orden { get; set; }
        public int TamanoColumna { get; set; }
        public bool? Deshabilitado { get; set; }
        public string? Grupo { get; set; }
        public int? Fila { get; set; }
        public int? IdIcono { get; set; }
        public bool? MostrarDashboard { get; set; }

        public virtual Dominio IdDominioNavigation { get; set; } = null!;
        public virtual Icono? IdIconoNavigation { get; set; }
        public virtual Seccion IdSeccionNavigation { get; set; } = null!;
        public virtual ICollection<EntidadEstructuraTablaValor> EntidadEstructuraTablaValor { get; set; }
    }
}
